src/components/ui/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { LinearProgress, LinearProgressProps } from "react-native-elements";
+import { Text, View } from "react-native";
+import { useStyles } from "./styles";
+import { colors } from "@app/utils";
+
+interface ProgressBarProps
+  extends Omit<LinearProgressProps, "value" | "color" | "variant"> {
+  id: string;
+  type: "info" | "warning" | "success";
+  maxValue: number;
+  curValue?: number;
+  label?: string;
+  hasProgressText?: boolean; // show progress in label
+  labelType?: "info" | "warning" | "success" | "default";
+}
+
+const colorMap = {
```

Comment:
maybe this could be moved to utils

Reply:
Hmm true. I think I used a color map in the button component as well but with different mappings. We need to decide what's our primary, secondary, info, danger and warning colours to be consistent